lib/image_processing/vips/gravity.rb

```diff
@@ -1,27 +1,39 @@
 module ImageProcessing
   module Vips
-    module Gravity
-      class InvalidGravityValue < StandardError; end
-
-      module_function
-
-      def get(image, width, height, gravity)
-        top = image.width - width
-        left = image.height - height
-        values = case gravity
-                  when "Center"    then [(top / 2), (left / 2)]
-                  when "North"     then [(top / 2), 0]
-                  when "East"      then [top, (left / 2)]
-                  when "South"     then [(top / 2), left]
-                  when "West"      then [0, (left / 2)]
-                  when "NorthEast" then [top, 0]
-                  when "SouthEast" then [top,left]
-                  when "SouthWest" then [0, left]
-                  when "NorthWest" then [0, 0]
-                  else
-                    raise InvalidGravityValue
-                  end
-        values.map(&:abs)
+    class Gravity
+      Error = Class.new(StandardError)
+
+      COORDS = {
+        "Center"    => -> (left, top) { [left / 2, top / 2] },
+        "North"     => -> (left, top) { [left / 2, 0      ] },
+        "NorthEast" => -> (left, top) { [left,     0      ] },
+        "East"      => -> (left, top) { [left,     top / 2] },
+        "SouthEast" => -> (left, top) { [left,     top    ] },
+        "South"     => -> (left, top) { [left / 2, top    ] },
+        "SouthWest" => -> (left, top) { [0,        top    ] },
+        "West"      => -> (left, top) { [0,        top / 2] },
+        "NorthWest" => -> (left, top) { [0,        0      ] },
+      }
+
+      def self.get_coords(image, width, height, gravity)
+        new(gravity).get_coords(image, width, height)
+      end
+
+      attr_reader :gravity
+
+      def initialize(gravity)
+        raise Error, "invalid gravity value: #{gravity.inspect}" unless COORDS.key?(gravity)
```

Comment:
Since gravity is case sensitive then it would be good to mention it in the error to reduce support. Also, printing valid keys for gravity in error is a really nice touch and makes for good UX.

Reply:
Thanks, I added the valid values in the error message (which should also automatically hint them about case sensitivity).